plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClientModule.java

```diff
@@ -76,11 +76,10 @@ public static Properties getConnectionProperties(MySqlConfig mySqlConfig)
         connectionProperties.setProperty("tinyInt1isBit", "false");
         connectionProperties.setProperty("rewriteBatchedStatements", "true");
 
-        // Try to make MySQL timestamps work (See https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-time-instants.html)
-        // without relying on server time zone (which may be configured to be totally unusable).
-        // TODO (https://github.com/trinodb/trino/issues/15668) rethink how timestamps are mapped. Also, probably worth adding tests
-        //  with MySQL server with a non-UTC system zone.
-        connectionProperties.setProperty("connectionTimeZone", "UTC");
+        // Force the connection time zone of that to the server to preserve time zones
```

Comment:
sounds opposite of what MySQL docs say:

according to docs: `connectionTimeZone = LOCAL` means that JDBC driver uses the JVM zone as the session zone.
`forceConnectionTimeZoneToSession = true` means that the server side connection zone is changed to match local JVM zone.

Now in addition to this I started thinking about whether `forceConnectionTimeZoneToSession = true` can lead to failure if the JVM zone doesn't exist on the server. This probably explains the existing comment about `without relying on server time zone (which may be configured to be totally unusable).`.

Reply:
see https://github.com/trinodb/trino/pull/15670 as an example of an issue I'm thinking about

Reply:
Yes, you're right. Comment is now backwards; I didn't update it from a previous attempt to get this to work. I'll update it to reflect what is in the documentation.